Library/Formula/node.rb

```diff
@@ -73,6 +73,10 @@ def install
       cd buildpath/"npm_install" do
         system "./configure", "--prefix=#{libexec}/npm"
         system "make", "install"
+        # `package.json` has relative paths to the npm_install directory.
+        # This copies back over the vanilla `package.json` that is expected.
+        # https://github.com/Homebrew/homebrew/issues/46131#issuecomment-157845008
+        cp buildpath/"npm_install/package.json", libexec/"npm/lib/node_modules/npm"
```

Comment:
Poking around a bit it looks like the problem is that if you install `npm` from a local destination, `npm` expects that local destination to stick around indefinitely and act as a sort of install/update base for `npm`. I'm not convinced in some way this isn't an edge-case bug that upstream should look at.

The hardcoded paths themselves don't seem to be a huge problem - They exist if you `npm install npm@latest -g` as well, albeit from whoever packaged that `npm` release upstream rather than a local user. Ref: https://github.com/Homebrew/homebrew/issues/46131#issuecomment-157855837

I'm happy here. I can't make it break via normal usage, or inside brew usage, and it resolves the presented issue. I'd like `npm` to take a look at this sort of setup process upstream, but I'm happy to merge this as-is.